packages/eslint-rules/lib/rules/no-custom-color.js

```diff
@@ -0,0 +1,174 @@
+const tailwindConfig = require("../../../shared/tailwind.config");
+const alwaysValidClasses = [
+  "text-base",
+  "text-center",
+  "text-2xl",
+  "text-left",
+  "text-lg",
+  "text-xl",
+  "bg-none",
+];
+
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description: "Incorrect command class name",
+      suggestion: false,
+    },
+    fixable: "code",
+  },
+  create: (context) => {
+    const targetClassNames = ["bg-", "text-"];
+    const findTargetClassNames = (classNames) => {
+      if (classNames === undefined) {
+        return;
+      }
+
+      if (
+        !Array.isArray(classNames) &&
+        (typeof classNames === "string" || classNames instanceof String)
+      ) {
+        classNames = classNames.replace(/\s{2,}/g, " ");
+        classNames = classNames.split(" ").filter((cls) => cls.length);
+      }
+
+      if (!Array.isArray(classNames)) {
+        return;
+      }
+
+      return classNames.filter((className) => {
+        return targetClassNames.some((matchClassName) =>
+          className.startsWith(matchClassName)
+        );
+      });
+    };
+
+    const mapValidClassNames = (colors, prefix = "") => {
+      return Object.keys(colors).reduce((r, k) => {
+        if (typeof colors[k] === "object" && colors[k] !== null) {
+          const newPrefix = prefix ? `${prefix}-${k}` : k;
+          r = [
+            ...new Set(r.concat(r, mapValidClassNames(colors[k], newPrefix))),
+          ];
+        } else {
+          r.push(`${prefix && `${prefix}-`}${k}`);
+        }
+        return r;
+      }, []);
```

Comment:
This feels quite hard to read for me. I don't feel like I understand 100% this code